middlewares/osio/auth_test.go

```diff
@@ -326,3 +327,40 @@ func TestRemoveUserID(t *testing.T) {
 		assert.Equal(t, "http://f8osoproxy.com/api/v1/namespaces/osio-ci-ee1-preview-che/pods?labelSelector=che.workspace_id%3Dworkspacew9zk6m4xggf0pbtk", req.URL.String())
 	})
 }
+
+func TestStripPathPrefix(t *testing.T) {
+	tables := []struct {
+		reqType      RequestType
+		reqPath      string
+		expectedPath string
+	}{
+		{api, "/api/api/anything", "/api/anything"},
+		{api, "/api/anything", "/api/anything"},
+		{oapi, "/oapi/oapi/anything", "/oapi/anything"},
```

Comment:
I wouldn't consider oapi in it self a separate path, it belongs to api.

When client is configured to use the /api prefix you'd get /api/api and /api/oapi